userguide/cpp.md

```diff
@@ -0,0 +1,198 @@
+# Using the C++ API
+
+Turi Create provides a C++ API to enable embedding into other applications or services, as a way to deploy models for inference, or to enable training in embedded scenarios where there is no Python interpreter. This consists of a set of headers; when those headers are consumed, the resulting library or application can be linked against the `libunity` provided by the `turicreate` package.
+
+The [API documentation for the C++ API](https://apple.github.io/turicreate/docs/cpp/) provides a more complete reference of available functions, classes, and methods. Note that this API is not stable and may change at any time; plan to pin a particular version of Turi Create to your application or service to make use of this API.
+
+##### Setup
+
+Using the C++ API requires building Turi Create from source, starting with a clone of the [repository](git@github.com:apple/turicreate.git). You should consider checking out a particular version tag (for instance `v4.1`) in order to keep API stability as you proceed (the API may change in master). Before proceeding to use the C++ API, run `./configure` to set up the build system for the project, and run `make -j8` from within `release/src/unity` to build the library artifacts that your application will depend on.
```

Comment:
I would say "release" rather than "version tag". I would also link to our main release page (https://github.com/apple/turicreate/releases) rather than "(for instance `v4.1`)".